packages/react-native-editor/__device-tests__/gutenberg-editor-history.test.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Internal dependencies
+ */
+import { blockNames } from './pages/editor-page';
+import { selectTextFromElement } from './helpers/utils';
+import testData from './helpers/test-data';
+
+it( 'should preserve editor history when a link has been added and configured to open in a new tab', async () => {
+	await editorPage.addNewBlock( blockNames.paragraph );
+	const paragraphBlockElement = await editorPage.getTextBlockAtPosition(
+		blockNames.paragraph
+	);
+
+	// Type text into block
+	await editorPage.typeTextToTextBlock(
+		paragraphBlockElement,
+		testData.shortText
+	);
+
+	// Select text
+	// Note: may need to write getParagraphBlockTextInputAtPosition helper to get the TextInput element
+	// Todo: how to select one single word of text from the input?
+	await selectTextFromElement( editorPage.driver, paragraphBlockElement );
+
+	// Tap link button in bottom bar
+
+	// Type link URL into input
+
+	// Tap "Apply"
+
+	// Tap "Open in new tab" switch
+
+	// Dismiss bottom sheet
+	await editorPage.dismissBottomSheet();
+
+	// Wait for the modal to close
+	await editorPage.driver.sleep( 3000 );
+
+	// Verify the link is created with correct attributes in the HTML with an assertion
+
+	// Type more text
+	await editorPage.typeTextToTextBlock(
+		paragraphBlockElement,
+		testData.mediumText
+	);
+
+	// Verify the link attributes have not shifted with an assertion
+
+	// Test undo button
+	// Test undo button
+	// Test undo button
+
+	// Test redo button
+	// Test redo button
+	// Test redo button
+
+	// Verify the link attributes have not shifted with an assertion
```

Comment:
To be sure I understand correctly, the focus of the test is to assert the entire history is re-playable, not attributes shifting, correct?

Reply:
You are correct -- the assertion of the replayable history should be the focus of the test. The shifting attributes are a symptom of the issue that we could also check for with an assertion, but this may not be the best place for it.